extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceRuntimeConfig.java

```diff
@@ -124,4 +125,16 @@
      */
     @ConfigItem
     public Optional<String> validationQuerySql;
+
+    /**
+     * The socket factory driver that should be appended as a datasource url property
+     */
+    @ConfigItem
+    public Optional<String> socketFactory;
```

Comment:
This can be removed since the configuration is set as a JDBC property and could live in the additionalJdbcProperties Map below